apps/barista-design-system/src/pages/single-page/components/icon-overview-content/icon-overview-content.ts

```diff
@@ -87,12 +92,31 @@ export class BaIconOverviewContent implements OnInit, AfterViewInit, OnDestroy {
       .subscribe((value: string) => {
         this._filteredIcons = filterIcons(this.icons, value);
         // remove or add the query param if we have a filter value
-        const queryParams = value.length ? { iconFilter: value } : {};
-
-        this._router.navigate([], {
-          queryParams,
-          relativeTo: this._activatedRoute,
-        });
+        if (this._platform.isBrowser) {
+          if (value.length) {
+            // we need to push the query change state without calling the angulars
+            // router navigate, because navigating would trigger the angular-routers
+            // scrollPositionRestoration, which will remove the focus from the current
+            // element and scroll back to top.
+            window.history.pushState(
+              {},
+              '',
+              `${window.location.origin}${window.location.pathname}`,
```

Comment:
I guess the value should be added as a parameter in the case its not empty - so it would need to be inverted. 

```suggestion
              `${window.location.origin}${window.location.pathname}?iconFilter=${value}`,
```